src/test/java/com/feedzai/commons/sql/abstraction/dml/result/ResultIteratorTest.java

```diff
@@ -0,0 +1,57 @@
+package com.feedzai.commons.sql.abstraction.dml.result;
+
+import com.feedzai.commons.sql.abstraction.engine.DatabaseEngineException;
+import mockit.Expectations;
+import mockit.Mocked;
+import org.junit.Test;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Class for testing the result iterator behaviour.
+ *
+ * @author Helena Poleri (helena.poleri@feedzai.com)
+ * @since 2.7.1
+ */
+public class ResultIteratorTest {
+
+    @Test
+    public void doesRowCountIncrementTest(@Mocked Statement statement,
+                                          @Mocked ResultSet resultSet)
+            throws DatabaseEngineException, SQLException {
+
+        MySqlResultIterator resultIterator = new MySqlResultIterator(statement, null);
```

Comment:
Use the `ResultIterator` on the variable declaration there's no need to be Mysql specific.
Also the variable can be final.